src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -280,46 +306,45 @@ internal void Serialize(Utf8JsonWriter writer)
             {
-                DefaultCertificate = CertificateManager.Instance.ListCertificates(StoreName.My, StoreLocation.CurrentUser, isValid: true, requireExportable: false)
-                    .FirstOrDefault();
-
-                if (DefaultCertificate != null)
-                {
-                    var status = CertificateManager.Instance.CheckCertificateState(DefaultCertificate, interactive: false);
-                    if (!status.Success)
-                    {
-                        // Display a warning indicating to the user that a prompt might appear and provide instructions on what to do in that
-                        // case. The underlying implementation of this check is specific to Mac OS and is handled within CheckCertificateState.
-                        // Kestrel must NEVER cause a UI prompt on a production system. We only attempt this here because Mac OS is not supported
-                        // in production.
-                        Debug.Assert(status.FailureMessage != null, "Status with a failure result must have a message.");
-                        logger.DeveloperCertificateFirstRun(status.FailureMessage);
-
-                        // Prevent binding to HTTPS if the certificate is not valid (avoid the prompt)
-                        DefaultCertificate = null;
-                    }
-                    else if (!CertificateManager.Instance.IsTrusted(DefaultCertificate))
-                    {
-                        logger.DeveloperCertificateNotTrusted();
-                    }
-                }
-                else
-                {
-                    logger.UnableToLocateDevelopmentCertificate();
-                }
+                logger.UnableToLocateDevelopmentCertificate();
+                return null;
             }
-            catch
+
+            var status = CertificateManager.Instance.CheckCertificateState(cert, interactive: false);
+            if (!status.Success)
             {
-                logger.UnableToLocateDevelopmentCertificate();
+                // Display a warning indicating to the user that a prompt might appear and provide instructions on what to do in that
+                // case. The underlying implementation of this check is specific to Mac OS and is handled within CheckCertificateState.
+                // Kestrel must NEVER cause a UI prompt on a production system. We only attempt this here because Mac OS is not supported
+                // in production.
+                Debug.Assert(status.FailureMessage != null, "Status with a failure result must have a message.");
+                logger.DeveloperCertificateFirstRun(status.FailureMessage);
+
+                // Prevent binding to HTTPS if the certificate is not valid (avoid the prompt)
+                return null;
             }
+
+            if (!CertificateManager.Instance.IsTrusted(cert))
+            {
+                logger.DeveloperCertificateNotTrusted();
+                return null;
```

Comment:
Behavior change, before it would log, but still use the dev cert / DefaultCertificate.
https://github.com/dotnet/aspnetcore/blob/dfd4cdea4d8d993ba96655927f59aa66cf7013cd/src/Servers/Kestrel/Core/src/KestrelServerOptions.cs#L309-L312

return the cert here.

Reply:
Huh.  I missed that - I definitely thought untrusted was a failure condition.

Reply:
browsers will give you a warning for untrusted certs, but will still let you access the site.